### PR TITLE
[Fix] Fix checkpoint local files detect

### PR DESCRIPTION
**Pull request by teamwong111**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Related Issue: https://github.com/open-mmlab/mmcv/issues/990
Detect checkpoint files more effectively.

## Modification

Add support for local files with relative path detect.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: tests/test_runner/test_checkpoint.py [hunk 1]
```diff
@@ -430,3 +431,15 @@ def test_save_checkpoint(tmp_path):
         save_checkpoint(
             model, filename, file_client_args={'backend': 'petrel'})
     mock_method.assert_called()
+
+
+def test_load_from_local():
+    import os
+    home_path = os.path.expanduser('~')
+    checkpoint_path = os.path.join(home_path, 'checkpoint.pth')
```
Comment (zhouzaida) [inline tests/test_runner/test_checkpoint.py:439]:
```suggestion
    checkpoint_path = os.path.join(home_path, 'dummy_checkpoint_used_to_test_load_from_local.pth')
```

File: tests/test_runner/test_checkpoint.py [hunk 2]
```diff
@@ -430,3 +431,15 @@ def test_save_checkpoint(tmp_path):
...
+    model = Model()
+    save_checkpoint(model, checkpoint_path)
+    checkpoint = load_from_local('~/checkpoint.pth', map_location=None)
```
Comment (zhouzaida) [inline tests/test_runner/test_checkpoint.py:442]:
```suggestion
    checkpoint = load_from_local('~/dummy_checkpoint_used_to_test_load_from_local.pth', map_location=None)
```

File: mmcv/runner/checkpoint.py [hunk 3]
```diff
@@ -260,7 +260,7 @@ def load_from_local(filename, map_location):
     Returns:
         dict or OrderedDict: The loaded checkpoint.
     """
-
+    filename = osp.expanduser(filename)
```
Comment (zhouzaida) [inline mmcv/runner/checkpoint.py:263]: line #265 can be changed to `raise FileNotFoundError(f'{filename} can not be found.')`

File: tests/test_runner/test_checkpoint.py (hunk 1 above)
Comment (teamwong111) [reply tests/test_runner/test_checkpoint.py]: Done

File: tests/test_runner/test_checkpoint.py (hunk 2 above)
Comment (teamwong111) [reply tests/test_runner/test_checkpoint.py]: Done

File: mmcv/runner/checkpoint.py (hunk 3 above)
Comment (teamwong111) [reply mmcv/runner/checkpoint.py:263]: Done

**Review by zhouzaida — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**